This removes ldflag -L"/usr/lib" and -Wl,-rpath=/usr/lib:/usr/lib which
causes `ld: skipping incompatible /usr/lib/libm.so when searching for -lm`
Reference: https://github.com/justxi/rocm/issues/8#issuecomment-1166193820
===================================================================
--- HIP-rocm-5.3.0/bin/hipcc.pl~	2022-10-14 15:53:24.000000000 +0100
+++ HIP-rocm-5.3.0/bin/hipcc.pl	2022-10-14 15:59:03.811408710 +0100
@@ -713,9 +713,9 @@
 
     if (not $isWindows  and not $compileOnly) {
       if ($linkType eq 0) {
-        $toolArgs = " -L$HIP_LIB_PATH -lamdhip64 -L$ROCM_PATH/lib -lhsa-runtime64 -ldl -lnuma " . ${toolArgs};
+        $toolArgs = " -lamdhip64 -lhsa-runtime64 -ldl -lnuma " . ${toolArgs};
       } else {
-        $toolArgs = ${toolArgs} . " -Wl,--enable-new-dtags -Wl,-rpath=$HIP_LIB_PATH:$ROCM_PATH/lib -lamdhip64 ";
+        $toolArgs = ${toolArgs} . " -Wl,--enable-new-dtags -lamdhip64 ";
       }
       # To support __fp16 and _Float16, explicitly link with compiler-rt
       $HIP_CLANG_BUILTIN_LIB="$HIP_CLANG_PATH/../lib/clang/$HIP_CLANG_VERSION/lib/$HIP_CLANG_TARGET/libclang_rt.builtins.a";
